src/collector/ObolCollectorFactory.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.19;
+import {LibClone} from "solady/utils/LibClone.sol";
+import {ObolCollector} from "./ObolCollector.sol";
+
+/// @title ObolCollector
+/// @author Obol
+/// @notice A factory contract for cheaply deploying ObolCollector.
+/// @dev The address returned should be used to as reward address collecting rewards
+contract ObolCollectorFactory {
+    /// -----------------------------------------------------------------------
+    /// errors
+    /// -----------------------------------------------------------------------
+
+    /// @dev Invalid address
+    error Invalid_Address();
+
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+    using LibClone for address;
+
+    /// -----------------------------------------------------------------------
+    /// events
+    /// -----------------------------------------------------------------------
+
+    /// Emitted on createCollector
+    event CreateCollector(address token, address withdrawalAddress);
+
+
+    /// -----------------------------------------------------------------------
+    /// storage
+    /// -----------------------------------------------------------------------
+
+    /// @dev collector implementation
+    ObolCollector public immutable collectorImpl;
```

Comment:
This can be `ISplitWalletImplementation` and `createCollector` can be virtual to be able to implement any type of `SplitWalletImplementation` contract

Also, on the same topic the `ObolCollector.sol` can be used as a base contract for any `SplitWalletImplementation` contracts